Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/theme/Theme.kt

```diff
@@ -17,60 +17,104 @@
+    outline = md_theme_light_outline,
+    inverseSurface = md_theme_light_inverseSurface,
+    inverseOnSurface = md_theme_light_inverseOnSurface,
+    inversePrimary = md_theme_light_inversePrimary,
+    surfaceTint = md_theme_light_surfaceTint,
 )
 
-val DarkThemeColors = darkColors(
-    primary = Purple300,
-    primaryVariant = Purple600,
-    onPrimary = Color.Black,
-    secondary = Color.Black,
-    onSecondary = Color.White,
-    background = Color.Black,
-    onBackground = Color.White,
-    surface = Color.Black,
-    onSurface = Color.White,
-    error = Red300,
-    onError = Color.Black
+private val DarkColors = darkColorScheme(
+    primary = md_theme_dark_primary,
+    onPrimary = md_theme_dark_onPrimary,
+    primaryContainer = md_theme_dark_primaryContainer,
+    onPrimaryContainer = md_theme_dark_onPrimaryContainer,
+    secondary = md_theme_dark_secondary,
+    onSecondary = md_theme_dark_onSecondary,
+    secondaryContainer = md_theme_dark_secondaryContainer,
+    onSecondaryContainer = md_theme_dark_onSecondaryContainer,
+    tertiary = md_theme_dark_tertiary,
+    onTertiary = md_theme_dark_onTertiary,
+    tertiaryContainer = md_theme_dark_tertiaryContainer,
+    onTertiaryContainer = md_theme_dark_onTertiaryContainer,
+    error = md_theme_dark_error,
+    onError = md_theme_dark_onError,
+    errorContainer = md_theme_dark_errorContainer,
+    onErrorContainer = md_theme_dark_onErrorContainer,
+    background = md_theme_dark_background,
+    onBackground = md_theme_dark_onBackground,
+    surface = md_theme_dark_surface,
+    onSurface = md_theme_dark_onSurface,
+    surfaceVariant = md_theme_dark_surfaceVariant,
+    onSurfaceVariant = md_theme_dark_onSurfaceVariant,
+    outline = md_theme_dark_outline,
+    inverseSurface = md_theme_dark_inverseSurface,
+    inverseOnSurface = md_theme_dark_inverseOnSurface,
+    inversePrimary = md_theme_dark_inversePrimary,
+    surfaceTint = md_theme_dark_surfaceTint,
 )
 
-val Colors.snackbarAction: Color
-    @Composable
-    get() = if (isLight) Purple300 else Purple700
-
-val Colors.progressIndicatorBackground: Color
-    @Composable
-    get() = if (isLight) Color.Black.copy(alpha = 0.12f) else Color.White.copy(alpha = 0.24f)
-
 @Composable
-fun JetsurveyTheme(darkTheme: Boolean = isSystemInDarkTheme(), content: @Composable() () -> Unit) {
-    val colors = if (darkTheme) {
-        DarkThemeColors
+fun JetsurveyTheme(
+    useDarkTheme: Boolean = isSystemInDarkTheme(),
+    content: @Composable () -> Unit
+) {
+    val colors = if (!useDarkTheme) {
+        LightColors
     } else {
-        LightThemeColors
+        DarkColors
+    }
+
+    val systemUiController = rememberSystemUiController()
+    DisposableEffect(systemUiController, useDarkTheme) {
+        systemUiController.setSystemBarsColor(
+            color = colors.surface,
+            darkIcons = !useDarkTheme
+        ) {
+            md_theme_dark_surface
```

Comment:
I think it might be better to leave this out, having dug into the Accompanist API a bit. WDYT?

Reply:
I was originally calling the system APIs directly, but I thought using Accompanist would give 3p devs more exposure to it. Jetsurvey (before any of these changes) has a bug where the status bar content color is too light in light mode, so it's unreadable. I just fixed it in this PR since I was matching everything to the design anyway.

Do you think there's a better alternative?

Reply:
Oh I meant just the color within the lambda — its default value should suffice here (I think)

Reply:
Ah, got it; yeah I'll get rid of the lambda